### DISCUSSION TITLE
OpSo model compatibility with other software

### DISCUSSION THREAD

**Discussion by canihaveabravo** [kitzeslab/opensoundscape#844] [Q&A]
Hello, 

I would be interested to import my trained model to Pamguard or Raven to run in real-time or to assist with offline analysis. Is a CNN trained in OpSo compatible with other software? 

Thanks, 
Lorenzo

**Comment by sammlapp** [comment]
Hi, we do not currently provide explicit integration with other software. Since OpenSoundscape uses PyTorch under the hood, other software that support PyTorch models can be used for down-stream analysis. (For instance, Pamguard seems to support PyTorch models while Raven does not). 

We have not tested importing PyTorch models into PamGuard, but be aware that you will need to specify the series of preprocessing operations (including spectrogram creation) in the PamGuard interface after loading the model. Since PamGuard expects a jit-compatible saved model file, you could save the opensoundscape model like this:
```
import torch
from opensoundscape import CNN

opso_cnn=CNN('resnet18',[0,1],2)
jit_model = torch.jit.script(opso_cnn.network)
jit_model.save("jit_model.pt")
```

I imagine it will take a bit of fiddling with the PamGuard interface to get this up and running. 

If you want to pursue this direction, Louis or I can assist you.

**Reply by canihaveabravo** [reply]
Hi there, I would like to pursue this option. Pamguard expect a tensorflow or a pytorch saved model file. I tried the code above to create a jit model but the file type .pt is not recognised. Can you assist please? thanks

**Reply by sammlapp** [reply]
I am unable to find documentation from Pamguard on loading PyTorch or Tensorflow models, and how the models should be saved. We'll need some documentation from PamGuard to proceed
